src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java

```diff
@@ -1,179 +1,338 @@
+                ShortCircuitConstants.DEFAULT_WITH_LIMIT_VIOLATIONS,
+                ShortCircuitConstants.DEFAULT_WITH_VOLTAGE_RESULT,
+                ShortCircuitConstants.DEFAULT_WITH_FORTESCUE_RESULT,
+                ShortCircuitConstants.DEFAULT_WITH_FEEDER_RESULT,
+                ShortCircuitConstants.DEFAULT_STUDY_TYPE,
+                ShortCircuitConstants.DEFAULT_MIN_VOLTAGE_DROP_PROPORTIONAL_THRESHOLD,
+                ShortCircuitPredefinedConfiguration.ICC_MAX_WITH_CEI909,
+                ShortCircuitConstants.DEFAULT_WITH_LOADS,
+                ShortCircuitConstants.DEFAULT_WITH_SHUNT_COMPENSATORS,
+                ShortCircuitConstants.DEFAULT_WITH_VSC_CONVERTER_STATIONS,
+                ShortCircuitConstants.DEFAULT_WITH_NEUTRAL_POSITION,
+                ShortCircuitConstants.DEFAULT_INITIAL_VOLTAGE_PROFILE_MODE
+                //ShortCircuitConstants.DEFAULT_SUB_TRANSIENT_COEFFICIENT
+                //ShortCircuitConstants.DEFAULT_DETAILED_REPORT
+        ));
+    }
 
-        @Override
-        public String getVersion() {
-            return "1.0";
-        }
+    @Test
+    void testCreateDefaultParameters() {
+        final UUID pUuid = UUID.randomUUID();
+        final AnalysisParametersEntity pEntity = spy(AnalysisParametersEntity.builder().id(pUuid).build());
+        when(parametersRepository.save(any(AnalysisParametersEntity.class))).thenReturn(pEntity);
+        assertThat(shortCircuitService.createParameters(null)).as("service call result").isEqualTo(pUuid);
+        verify(pEntity).getId();
+        final ArgumentCaptor<AnalysisParametersEntity> captor = ArgumentCaptor.forClass(AnalysisParametersEntity.class);
+        verify(parametersRepository).save(captor.capture());
+        assertThat(captor.getValue()).usingRecursiveComparison().isEqualTo(new AnalysisParametersEntity());
+    }
 
-        @Override
-        public CompletableFuture<ShortCircuitAnalysisResult> run(Network network, List<Fault> faults, ShortCircuitParameters parameters, ComputationManager computationManager, List<FaultParameters> faultParameters) {
-            return CompletableFuture.completedFuture(this.result);
-        }
+    @Test
+    void testUpdateNonExistingParameters() {
+        final UUID pUuid = UUID.randomUUID();
+        when(parametersRepository.findById(any(UUID.class))).thenReturn(Optional.empty());
+        assertThat(shortCircuitService.updateOrResetParameters(pUuid, null)).as("service call result").isFalse();
+        verify(parametersRepository).findById(pUuid);
+    }
 
-        @Override
-        public CompletableFuture<ShortCircuitAnalysisResult> run(Network network, List<Fault> faults, ShortCircuitParameters parameters, ComputationManager computationManager, List<FaultParameters> faultParameters, ReportNode reportNode) {
-            return CompletableFuture.completedFuture(this.result);
-        }
+    @Test
+    void testUpdateExistingParameters() {
```

Comment:
In the end what do you want to test ? There are too many checks here, it feels like not all of them bring value

Reply:
I add comments to explain, and I de-duplicate some code.